controller.cpp

```diff
@@ -66,6 +83,48 @@ void Ctrl::notifyUser(std::string str){
   gui->appendOutput(str);
 }
 
-void Ctrl::notifyUserUnicode(std::wstring wstr){
+void Ctrl::notifyUser(std::wstring wstr){
   gui->appendOutput(wstr);
 }
+
+std::wstring Ctrl::generateChemicalFormulaUnicode(std::string chemical_formula){
+  std::wstring chemical_formula_unicode;
+  for (int i = 0; i < chemical_formula.size(); i++){
+    if (isalpha(chemical_formula[i])){
+      // it is much simpler to convert string to wxString to wstring than directly from string to wstring
+      wxString buffer(chemical_formula[i]);
+      chemical_formula_unicode.append(buffer.wxString::ToStdWstring());
+    }
+    else if (chemical_formula[i] == '0'){
+      chemical_formula_unicode.append(L"\u2080");
+    }
+    else if (chemical_formula[i] == '1'){
+      chemical_formula_unicode.append(L"\u2081");
+    }
+    else if (chemical_formula[i] == '2'){
+      chemical_formula_unicode.append(L"\u2082");
+    }
+    else if (chemical_formula[i] == '3'){
+      chemical_formula_unicode.append(L"\u2083");
+    }
+    else if (chemical_formula[i] == '4'){
+      chemical_formula_unicode.append(L"\u2084");
+    }
+    else if (chemical_formula[i] == '5'){
+      chemical_formula_unicode.append(L"(\u2085");
+    }
+    else if (chemical_formula[i] == '6'){
+      chemical_formula_unicode.append(L"\u2086)");
+    }
+    else if (chemical_formula[i] == '7'){
+      chemical_formula_unicode.append(L"\u2087");
+    }
+    else if (chemical_formula[i] == '8'){
+      chemical_formula_unicode.append(L"\u2088");
+    }
+    else if (chemical_formula[i] == '9'){
+      chemical_formula_unicode.append(L"\u2089");
+    }

```

Comment:
As a general rule, you should never have to copy-paste code more than twice. This can be written in a single line.

Reply:
I wasn't sure if I could make a loop that would include different wstring values. If you want to enlighten me, be my guest.

Reply:
Ok, I haven't quite gotten it into one line, but I've found a way to take a whole string of numbers and convert it to a string of corresponding subscripts without using wstring.